__tests__/app.test.js

```diff
@@ -35,14 +34,88 @@ describe("GET /api/topics", () => {
-    test("200: returns a json object", () => {
-       return request(app)
-       .get('/api')
-       .expect(200)
-       .then((response) => {
+  test("200: returns a json object", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((response) => {
         expect(response.body.endpoints).toEqual(endpoints);
-       })
       });
-});
\ No newline at end of file
+  });
+});
+describe("/api/articles/:article_id", () => {
+  test("GET: 200 sends an article by its id", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.article.article_id).toBe(1);
+        expect(response.body.article.title).toBe("Living in the shadow of a great man");
+        expect(response.body.article.topic).toBe("mitch");
+        expect(response.body.article.author).toBe("butter_bridge");
+        expect(response.body.article.body).toBe("I find this existence challenging");
+        expect(response.body.article.created_at).toBe("2020-07-09T20:11:00.000Z");
+        expect(response.body.article.votes).toBe(100);
+      });
+  });
+  test("GET:404 sends an appropriate status and error message when given a valid but non-existent id", () => {
+    return request(app)
+      .get("/api/articles/99")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("article does not exist");
+      });
+  });
+  test("GET:400 sends an appropriate status and error message when given an invalid id", () => {
+    return request(app)
+      .get("/api/articles/not-an-article")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad request");
+      });
+  });
+});
+describe("/api/articles", () => {
+  test("GET: 200 sends the articles and they should be sorted by date in descending order with no body propert present", () => {
+    return request(app)
+    .get('/api/articles')
+    .expect(200)
+    .then((response)=> {
+      const articles = response.body.articles
+      expect(articles).toHaveLength(13)
+      articles.forEach(article => {
+        expect(article).toMatchObject({
+        article_id: expect.any(Number),
+        author: expect.any(String),
+        title: expect.any(String),
+        topic: expect.any(String),
+        created_at: expect.any(String),
+        votes: expect.any(Number),
+        article_img_url: expect.any(String),
+        comment_count: expect.any(Number)
+      })
+      })
+    })
+  })
+  test("GET: 200 sends the articles sorted by date in descending order", () => {
+    return request(app)
+    .get("/api/articles")
+    .expect(200)
+    .then((response) => {
+      const articles = response.body.articles
+      expect(articles.at(0).article_id).not.toBe(1)
+      expect(articles.at(0).created_at).toBe("2020-11-03T09:12:00.000Z");
+    })
+  })
+  test("GET:404 responds with an appropriate status and error message when given a non-existent api", () => {
```

Comment:
This test isn't specific to the api/articles path, so can either be removed, or put into it's own describe block for incorrect paths. 👍